iroh-resolver/src/resolver.rs

```diff
@@ -119,18 +119,20 @@ impl Path {
         self.tail.push(str.as_ref().to_owned());
     }
 
-    pub fn to_string_without_type(&self) -> String {
-        let mut s = format!("{}", self.root);
-        for part in &self.tail {
-            if part.is_empty() {
-                continue;
-            }
-            s.push_str(&format!("/{}", part)[..]);
-        }
-        if self.has_trailing_slash() {
-            s.push('/');
+    // Arqu says that empty path segments in the *middle* shouldn't occur,
+    // though they can occur at the end, which `join` handles.
+    // TODO: it would make sense to return a `RelativePathBuf` here at some
+    // point in the future so we don't deal with bare strings anymore and
+    // we're forced to handle various cases more explicitly.
```

Comment:
Can you write the TODO as `TODO(faassen)`

I think, let's also take out the reference to asmir here, let's not get into a finger-pointing situation if an empty path segment does occur, lol